newsfragments/3197.feature.rst

```diff
@@ -0,0 +1 @@
+Add :func:`@trio.background_with_channel <trio.background_with_channel>`, a wrapper that can be used to make async generators safe. See :ref:`async-generators`, `ASYNC900 <https://flake8-async.readthedocs.io/en/latest/rules.html#async900>`_, :pep:`789`, and :pep:`533`.
```

Comment:
Maybe add a short summary of why each of these links are relevant instead of just having four in a row. Similar to:

```suggestion
Add :func:`@trio.background_with_channel <trio.background_with_channel>`, a wrapper that can be used to make async generators safe. See :ref:`a review on why they are unsafe <async-generators>`. This addition allows for straightforward recommendations for `ASYNC900 <https://flake8-async.readthedocs.io/en/latest/rules.html#async900>`_ and :pep:`789`.
```

(yes I cut out 533)

Reply:
the link to `async-generators` isn't really needed after the `autofunction` is moved to that section, and both it and async900 (now) link to pep789, so we can drop that too